// supermemory/resources/memories.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Memories = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
class Memories extends resource_1.APIResource {
    /**
     * Update a memory with any content type (text, url, file, etc.) and metadata
     *
     * @example
     * ```ts
     * const memory = await client.memories.update('id', {
     *   content:
     *     'This is a detailed article about machine learning concepts...',
     * });
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/v3/memories/${id}`, { body, ...options });
    }
    /**
     * Retrieves a paginated list of memories with their metadata and workflow status
     *
     * @example
     * ```ts
     * const memories = await client.memories.list();
     * ```
     */
    list(body = {}, options) {
        return this._client.post('/v3/memories/list', { body, ...options });
    }
    /**
     * Delete a memory
     *
     * @example
     * ```ts
     * await client.memories.delete('id');
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/v3/memories/${id}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Add a memory with any content type (text, url, file, etc.) and metadata
     *
     * @example
     * ```ts
     * const response = await client.memories.add({
     *   content:
     *     'This is a detailed article about machine learning concepts...',
     * });
     * ```
     */
    add(body, options) {
        return this._client.post('/v3/memories', { body, ...options });
    }
    /**
     * Get a memory by ID
     *
     * @example
     * ```ts
     * const memory = await client.memories.get('id');
     * ```
     */
    get(id, options) {
        return this._client.get((0, path_1.path) `/v3/memories/${id}`, options);
    }
}
exports.Memories = Memories;
